lib/guide.md

```diff
@@ -0,0 +1,1061 @@
+
+        System.out.println("Was transaction accepted? " + receipt.isAccepted() + ".");
+    }
+}
+```
+
+## Estimating fee for declare V2 transaction
+<!-- codeSection(path="starknet/account/StandardAccountTest.kt", function="estimateFeeForDeclareV2Transaction", language="kotlin") -->
+```java
+import com.swmansion.starknet.account.Account;
+import com.swmansion.starknet.account.StandardAccount;
+import com.swmansion.starknet.crypto.StarknetCurve;
+import com.swmansion.starknet.data.types.*;
+import com.swmansion.starknet.provider.Provider;
+import com.swmansion.starknet.provider.Request;
+import com.swmansion.starknet.provider.rpc.JsonRpcProvider;
+
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+
+public class Main {
+    public static void main(String[] args) throws IOException, InterruptedException {
+        // Create a provider for interacting with Starknet
+        Provider provider = new JsonRpcProvider("https://example-node-url.com/rpc");
+
+        // Set up an account
+        Felt privateKey = Felt.fromHex("0x1234");
+        Felt publicKey = StarknetCurve.getPublicKey(privateKey);
+        Felt accountAddress = Felt.fromHex("0x1236789");
+        // ⚠️ WARNING ⚠️ Both the account address and private key are for demonstration purposes only.
+
+        StarknetChainId chainId = provider.getChainId().send();
+        Account account = new StandardAccount(accountAddress, privateKey, provider, chainId, Felt.ZERO);
+
+        // Import a compiled contract
+        Path contractPath = Paths.get("contract.sierra.json");
+        Path casmPath = Paths.get("contract.casm.json");
+        String contractCode = String.join("", Files.readAllLines(contractPath));
+        String casmCode = String.join("", Files.readAllLines(casmPath));
+
+        Cairo1ContractDefinition contractDefinition = new Cairo1ContractDefinition(contractCode);
+        CasmContractDefinition casmContractDefinition = new CasmContractDefinition(casmCode);
+        Felt nonce = account.getNonce().send();
+
+        DeclareTransactionV2Payload payload = account.signDeclareV2(
+                contractDefinition,
+                casmContractDefinition,
+                new ExecutionParams(nonce,Felt.ZERO),
+                true
+        );
+        Request<EstimateFeeResponseList> request = provider.getEstimateFee(List.of(payload), Collections.emptySet());
+        EstimateFeeResponseList response = request.send();
+        EstimateFeeResponse feeEstimate = response.getValues().get(0);
+        
+        System.out.println("The estimated fee is: " + feeEstimate.getOverallFee().getValue() + ".");
+    }
+}
+```
+
+# Package com.swmansion.starknet.account
+Account interface used to simplify preparing, signing Starknet transactions and automatic fee estimation.
```

Comment:
Nit:
I think this can be simplified and include information about *sending* mentioned transactions for execution, as this seems one of most common uses for the interface 😄

Reply:
Also, I know this inherited from the previous version of the docs, but afaik we document the entire *package*, not just *interface*, so maybe we should include info on `StandardAccount` as well

Reply:
Sure, updated to 
```
[Account](src/main/kotlin/com/swmansion/starknet/account/Account.kt) interface is used to send Starknet transactions for execution. Its base implementation is [StandardAccount](src/main/kotlin/com/swmansion/starknet/account/StandardAccount.kt).
```
(both account interface and standard account class are linked) 👍 .